common/app/templates/inlineJS/nonBlocking/detectAdblock.scala.js

```diff
@@ -0,0 +1,28 @@
+try {
+    ((document, window) => {
+        window.guardian.adBlockers = {};
+        var ad = document.createElement('div');
+        ad.style.position = 'absolute';
+        ad.style.left = '-9999px';
+        ad.style.height = '10px';
+        ad.innerHTML = '&nbsp;';
+        ad.setAttribute('class', 'ad_unit');
+
+        // avoid a forced sync layout
+        window.requestAnimationFrame(() => {
+            document.body.appendChild(ad);
+
+            // avoid a forced layout, and be sure the element has been added to the DOM
+            window.requestAnimationFrame(() => {
+                var adStyles = window.getComputedStyle(ad);
+                window.guardian.adBlockers.generic = adStyles.getPropertyValue('display') === 'none';
+                window.guardian.adBlockers.ffAdblockPlus = adStyles.getPropertyValue('-moz-binding').match('elemhidehit') !== null;
+                try {
+                    window.guardian.adBlockers.onDetect();
```

Comment:
Super pedantic, but should this technically be an array of listeners? I guess I'm splitting hairs here...